exchange/price_granularity.go

```diff
@@ -13,14 +13,18 @@ func GetPriceBucket(cpm float64, config openrtb_ext.PriceGranularity) string {
 	bucketMax := 0.0
 	increment := 0.0
 	precision := config.Precision
-
+	bucketFloor := 0.0
+	bucketMin := 0.0
 	for i := 0; i < len(config.Ranges); i++ {
 		if config.Ranges[i].Max > bucketMax {
 			bucketMax = config.Ranges[i].Max
 		}
 		// find what range cpm is in
 		if cpm >= config.Ranges[i].Min && cpm <= config.Ranges[i].Max {
 			increment = config.Ranges[i].Increment
+			bucketMin = bucketFloor
+		} else {
+			bucketFloor = config.Ranges[i].Max
```

Comment:
You don't need to keep track of the prev max and assign that to `bucketMin`. This is already done [here](https://github.com/prebid/prebid-server/blob/master/openrtb_ext/request.go#L241). You can just simply keep track of the smallest `bucketMin`, similar to how `bucketMax` is tracked and pass that value to the `getCpmTarget` function. With this change you won't need `bucketFloor` at all.

Something like:

```
	bucketMax := 0.0
	bucketMin := math.MaxFloat64
	increment := 0.0
	precision := config.Precision

	for i := 0; i < len(config.Ranges); i++ {
		if config.Ranges[i].Min < bucketMin {
			bucketMin = config.Ranges[i].Min
		}
		if config.Ranges[i].Max > bucketMax {
			bucketMax = config.Ranges[i].Max
		}
```

Reply:
@mansinahar 
We want to keep track for the bucketMin  for the range from which increment is selected, not the smallest bucketMin. smallest bucketMin will always be zero as done [here](https://github.com/prebid/prebid-server/blob/f29682a87293010a346aeed4a5870b4fe4c37eee/openrtb_ext/request.go#L232).

Reply:
@Pubmatic-Dhruv-Sonone Ah! You're right. That was my mistake. So you only need bucketMin = config.Ranges[i].Min on L25 and we can remove bucketFloor altogether as the validation code already makes sure that the previous max is set as min for the next range. Can you please make that change? Thanks!

Reply:
Done. Made the change to set ` bucketMin = config.Ranges[i].Min`